### PR TITLE
fix(hybrid-cloud): Adds region fan-out to organizations list command

### PR DESCRIPTION
**Pull request by GabeVillalobos**
The `organization list` command currently relies on the base `https://sentry.io/api/0/organizations/` route to query the user's active organizations; however, this will stop working when we incorporate multiple regions.

In order to address this, we've added logic to query an active user's regions, then fan-out to each of the individual regions in order to query all active organizations. This list is then combined and sorted by name as per usual.

In order to make this change, some of the underlying URI and request construction code had to be modified to allow overriding the request hostname with the region's hostname.

### CODE REVIEW

File: src/api.rs [hunk 1]
```diff
@@ -1466,6 +1513,16 @@ impl Api {
         Ok(rv)
     }
 
+    pub fn list_available_regions(&self) -> ApiResult<Vec<Region>> {
+        let resp = self.get("/users/me/regions/")?;
+        if resp.status() == 404 || resp.status == 400 {
+            return Err(ApiErrorKind::ResourceNotFound.into());
+        }
```
Comment (markstory) [inline src/api.rs:1520]: If the response is a 404 should we return an empty list result? Someone could be using new sentry-cli with old self-hosted.

**Review by markstory — APPROVED** [review]
Looks good to me from a hybrid cloud compatibility point of view, but I'm no rust expert.

**Review by dashed — APPROVED** [review]

File: src/api.rs [hunk 2]
```diff
@@ -1466,6 +1513,22 @@ impl Api {
...
 
+    pub fn list_available_regions(&self) -> ApiResult<Vec<Region>> {
+        let resp = self.get("/users/me/regions/")?;
+        if resp.status() == 404 {
+            // This endpoint may not exist for self-hosted users, so
+            // returning a default of [] seems appropriate.
+            return Ok(vec![]);
+        }
+
+        if resp.status == 400 {
```
Comment (dashed) [inline src/api.rs:1524]:
```suggestion
        if resp.status() == 400 {
```

**Review by dashed — APPROVED** [review]

File: tests/integration/organizations/list.rs [hunk 3]
```diff
@@ -6,6 +8,21 @@ fn command_organizations_list() {
...
             .with_response_file("organizations/get-organizations.json"),
     );
+
+    let region_response = format!(
+        "{{
+            \"regions\": [{{
+                \"name\": \"monolith\",
+                \"url\": \"{}\"
+            }}]
+        }}",
```
Comment (loewenheim) [inline tests/integration/organizations/list.rs:18]: This is slightly nicer with a raw string.
```suggestion
        r#"{{
            "regions": [{{
                "name": "monolith",
                "url": "{}"
            }}]
        }}"#,
```

**Review by loewenheim — APPROVED** [review]

**Pull request merged**